src/MessagingAttachment/MessagingAttachment.tsx

```diff
@@ -12,27 +13,41 @@ function cssClass(element: string) {
 
 type Props = {
   attachmentID: string;
+  // TODO: should this be optional?
+  attachmentURL: string;
   errorMsg?: string;
-  icon: React.ReactNode;
+  // TODO: is this needed, or only mimeType?
+  fileType: AttachmentFileType;
+  icon?: React.ReactNode;
+  isUpload?: boolean;
+  mimeType?: string;
   onClickAttachment: (attachmentID: string) => void;
   onRemoveAttachment?: (attachmentID: string) => void;
-  title?: string;
   subtitle?: string;
-  isUpload?: boolean;
+  title?: string;
   uploadComplete?: boolean;
 };
 
+// TODO: should I use a discriminated union here? to keep the props neat?
```

Comment:
Seems not strictly necessary IMO, considering how much work you've already done here. That said, it might actually help with keeping things neat going forward and prevent breakage!